ui/src/lib/yjs/useYjsStore.ts

```diff
@@ -17,15 +18,30 @@ export default ({
   workflowId?: string;
   handleWorkflowIdChange: (id?: string) => void;
 }) => {
-  const [{ yWorkflows }] = useState(() => {
-    // TODO: setup middleware/websocket provider
-    const yDoc = new Y.Doc();
-    const yWorkflows = yDoc.getArray<YWorkflow>("workflows");
+  const yDocRef = useRef<Y.Doc | null>(null)
+  const yWebSocketRef = useRef<WebsocketProvider | null>(null);
+
+  const [{ yWorkflows }, setYWorkflows] = useState<{ yWorkflows: Y.Array<YWorkflow> }>({ yWorkflows: new Y.Array() });
+
+
+  useEffect(() => {
+    yDocRef.current = new Y.Doc()
+    yWebSocketRef.current = new WebsocketProvider(
+      "ws://localhost:8000",
+      workflowId ? workflowId : "",
+      yDocRef.current,
+      { params: { token: "nyaan" } },
+    )
+    const yWorkflows = yDocRef.current.getArray<YWorkflow>("workflows");
     const yWorkflow = yWorkflowBuilder("main", "Main Workflow");
     yWorkflows.push([yWorkflow]);
+    setYWorkflows({ yWorkflows });
 
-    return { yWorkflows };
-  });
+    return () => {
+      yDocRef.current?.destroy();
+      yWebSocketRef.current?.destroy();
+    }
+  }, [])
```

Comment:
Please check code rabbit review before asking for other's reviews. Sometimes it can catch small stuff like this that you can then fix up before a developer reads the PR.
Here you should be passing workflowId or else changing to a different project/workflow will use the same Ydoc